tensorboard/backend/empty_path_redirect.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Redirect from an empty path to "/".
```

Comment:
Can you clarify somewhere in the text here or the docstring below that the actual redirect issued is relative to the containing path, i.e. SCRIPT_NAME?  Otherwise it's a bit ambiguous how this behaves when it's not the actual root WSGI app.

(FWIW, making this entirely conceptually pure would be something like raising an exception to redirect to the path "/" and then letting the containing WSGI middlewares prepend their paths before actually redirecting instead of using SCRIPT_NAME, so this middleware is really 100% ignorant of its containing path.  I don't think we should actually do that because it's much more complicated that necessary, but just to illustrate the kind of questions that emerge which could be resolved by clarifying the docs.)

Reply:
Docs updated.

I don’t quite follow the “conceptually pure” approach. WSGI doesn’t
prescribe any standard exception types, so wouldn’t all the middlewares
in question need to coordinate about which exception to use and how to
interpret it? This seems like it requires _more_ coupling among
middlewares, not less, given that this is the intended purpose of
`SCRIPT_NAME` (afaik, but implied by [articles like this one][1]).

Agreed that this kind of question is worth answering in the docs; thanks
for the suggestion. :-)

[1]: http://lucumr.pocoo.org/2007/5/21/getting-started-with-wsgi/